plugins/pulp_docker/plugins/importers/sync.py

```diff
@@ -127,7 +127,8 @@ def add_v1_steps(self, repo, config):
         :param config:      config object for the sync
         :type  config:      pulp.plugins.config.PluginCallConfiguration
         """
-        self.add_child(v1_sync.GetMetadataStep())
+
+        self.add_child(v1_sync.GetMetadataStep(config=config))
```

Comment:
i am not really sure that would be applicable to v1 images. the scope of whitelistening the tags was just for v2 content.
https://docs.pulpproject.org/plugins/pulp_docker/tech-reference/tags.html

Reply:
hm, nevertheless i tested and it works.
```> db.repos.find({'repo_id':'v1synctest1'})
{ "_id" : ObjectId("5b212abf45ef4834df2c4f25"), "repo_id" : "v1synctest1", "notes" : { "_repo-type" : "docker-repo" }, "scratchpad" : { "tags" : [ { "image_id" : "c058d2db34b192789daa4c7c304f87f491b5602ca150328f726035e31efb3550", "tag" : "1.24.2-glibc" } ] }, "content_unit_counts" : { "docker_image" : 2 }, "_ns" : "repos", "last_unit_added" : ISODate("2018-06-13T14:42:43.049Z") }
> 
$ pulp-admin docker repo search image --repo-id v1synctest1
Warning: path should have mode 0700 because it may contain sensitive information: /home/ipanova/.pulp/

Created:      2018-06-13T14:42:43Z
Metadata:     
  Downloaded:         True
  Image Id:           5f7820582aae34a153bd540c057ba80f62e94ad5cfab23a1f9533df984
                      167748
  Pulp User Metadata: 
Repo Id:      v1synctest1
Unit Id:      26a700b4-f996-497f-94e9-341288fbe06a
Unit Type Id: docker_image
Updated:      2018-06-13T14:42:43Z

Created:      2018-06-13T14:42:43Z
Metadata:     
  Downloaded:         True
  Image Id:           c058d2db34b192789daa4c7c304f87f491b5602ca150328f726035e31e
                      fb3550
  Parent Id:          5f7820582aae34a153bd540c057ba80f62e94ad5cfab23a1f9533df984
                      167748
  Pulp User Metadata: 
  Tags:               1.24.2-glibc
Repo Id:      v1synctest1
Unit Id:      a393fd48-7901-4f40-bb83-e13806138fd1
Unit Type Id: docker_image
Updated:      2018-06-13T14:42:43Z

```

Reply:
I had to pass config through to access it in the v1 sync code. I think since it was really easy to get it working in v1 it doesn't hurt to leave it in. But I can strip it out if you think there's some reason why we shouldn't.

Reply:
it looks good, i just do not know what to do with the fact that:
once we want to get rid of some tag locally - we need to manually remove it and the referenced content.
Since in v1 tags are embedded - we cannot remove the tag itself but we'd need to directly remove the content which it was referencing.
so i went and removed the the parent_id with thinking that it will drag all it's children..and.. it it did not happen.
not sure if this is ^ a bug or desired behaviour but looks like user would need to manually remove each and every image_id.